src/Android/Services/AndroidPushNotificationService.cs

```diff
@@ -41,11 +41,13 @@ public Task<bool> AreNotificationsSettingsEnabledAsync()
 
         public async Task<string> GetTokenAsync()
         {
-            return await _stateService.GetPushCurrentTokenAsync();
+            var activeUserId = await _stateService.GetActiveUserIdAsync();
+            return await _stateService.GetPushCurrentTokenAsync(activeUserId);
         }
 
         public async Task RegisterAsync()
         {
+            var activeUserId = await _stateService.GetActiveUserIdAsync();
```

Comment:
The `activeUserId` is only used in the `else` so unless I'm missing something it's better to move this line inside the `else` to improve performance on the other condition

Reply:
I refactored to have a `null` default value for the state service methods, so this is no longer needed.